### PR TITLE
feat: accept solver timeout as parameter on CLI

### PR DESCRIPTION
**Pull request by carrascomj**
* [x] workaround #680 
`test_find_reactions_unbounded_flux_default_condition` in test_consistency can get stuck due to numerical instabilities while performing FVA.
* [x] Description of feature
Add a parameter on the CLI to allow the user to provide a solver timeout value.
```
memote --timeout-solver 3 run [...]
```
With this timeout, the test doesn't get stuck and continues. Whenever the solver reaches the timeout, it outputs an `UserWarning`. The problem is that timeout reactions won't be catch as Infeasible by the FVA implementation on cobrapy.

* [x] Decide to set a default value on timeout or temporarily set a value only for `test_find_reactions_unbounded_flux_default_condition` if none was provided.

* [x] tests added/passed
* [ ] add an entry to the [next release](../HISTORY.rst)

### CODE REVIEW

**Comment by Midnighter** [comment]
This would be really neat to make an attribute on the `cobrapy.Configuration` class, we could then set it here through the command line option.

**Comment by carrascomj** [comment]
On the cobrapy side, it would also be really helpful to slightly modify FVA to handle the timeout case. Right now, for every reaction, [the implementation only checks if the value is `None`](https://github.com/opencobra/cobrapy/blob/devel/cobra/flux_analysis/variability.py#L54) and that's not the case here, where the model.solver.status is set to `time_out` but the value of the solution is still a float.

**Comment by Midnighter** [comment]
Probably it's better discussed in an issue on cobrapy but I don't fully see your point. In the line above that one

```python
value = _model.solver.objective.value
```

if the solver experienced a time-out, this should be a valid value, shouldn't it?

**Comment by carrascomj** [comment]
Yes, it is a valid value but I am not sure if it should be considered a valid solution in the FVA loop or it should be discarded.

File: src/memote/suite/cli/runner.py [hunk 1]
```diff
@@ -111,6 +111,9 @@ def cli():
               type=click.Choice(["cplex", "glpk", "gurobi", "glpk_exact"]),
               default="glpk", show_default=True,
               help="Set the solver to be used.")
+@click.option("--timeout-solver",
+              type=int,
+              help="Timeout to put in the LP solver.")
```
Comment (Midnighter) [inline src/memote/suite/cli/runner.py]:
```suggestion
              help="Timeout in seconds to set on the mathematical optimization solver.")
```

**Comment by carrascomj** [comment]
I think the `test_model` test with a big model like you said would be problematic: the same test would potentially pass on Travis but fail on a local, faster, computer.

I believe that a better test would be to use a model that is known to get stuck on FVA. Another way is to set the timeout to 0 so another model wouldn't be needed.

File: src/memote/suite/cli/config.py [hunk 2]
```diff
@@ -51,7 +51,7 @@ class Memote(SectionSchema):
             type=click.Choice(["cplex", "glpk", "gurobi", "glpk_exact"]),
             default="glpk"
         )
-        timeout_solver = Param(type=int)
+        solver_timeou = Param(type=int, default=None)
```
Comment (Midnighter) [inline src/memote/suite/cli/config.py]:
```suggestion
        solver_timeout = Param(type=int, default=None)
```

File: src/memote/suite/cli/reports.py [hunk 3]
```diff
@@ -69,6 +69,10 @@ def report():
               type=click.Choice(["cplex", "glpk", "gurobi", "glpk_exact"]),
               default="glpk", show_default=True,
               help="Set the solver to be used.")
+@click.option("--solver_timeout",
```
Comment (Midnighter) [inline src/memote/suite/cli/reports.py]:
```suggestion
@click.option("--solver-timeout",
```

File: src/memote/suite/cli/reports.py [hunk 4]
```diff
@@ -204,6 +210,10 @@ def _test_diff(model_and_model_ver_tuple, pytest_args, skip,
 @click.option("--solver", type=click.Choice(["cplex", "glpk", "gurobi"]),
               default="glpk", show_default=True,
               help="Set the solver to be used.")
+@click.option("--solver_timeout",
```
Comment (Midnighter) [inline src/memote/suite/cli/reports.py]:
```suggestion
@click.option("--solver-timeout",
```

File: src/memote/suite/cli/runner.py [hunk 5]
```diff
@@ -111,9 +111,10 @@ def cli():
               type=click.Choice(["cplex", "glpk", "gurobi", "glpk_exact"]),
               default="glpk", show_default=True,
               help="Set the solver to be used.")
-@click.option("--timeout-solver",
-              type=int,
-              help="Timeout to put in the LP solver.")
+@click.option("--solver_timeout",
```
Comment (Midnighter) [inline src/memote/suite/cli/runner.py]:
```suggestion
@click.option("--solver-timeout",
```

File: src/memote/suite/cli/runner.py [hunk 6]
```diff
@@ -297,6 +298,10 @@ def _test_history(model, sbml_ver, solver, manager, commit, pytest_args, skip,
 @click.option("--solver", type=click.Choice(["cplex", "glpk", "gurobi"]),
               default="glpk", show_default=True,
               help="Set the solver to be used.")
+@click.option("--solver_timeout",
```
Comment (Midnighter) [inline src/memote/suite/cli/runner.py]:
```suggestion
@click.option("--solver-timeout",
```

**Comment by carrascomj** [comment]
I have been trying some things with the test but in the end I've just mark it as xfailed. Maybe the only option is to try to work with a very large model like you said @Midnighter.

**Pull request merged**